### PR TITLE
support both white and blacklist at the same time; issue-476

### PR DESCRIPTION
**Pull request by tpashkin**

### CODE REVIEW

File: cloud/storage/core/libs/features/features_config.h [hunk 1]
```diff
@@ -16,13 +16,13 @@ class TFeaturesConfig
...
     {
-        THashSet<TString> CloudIds;
-        THashSet<TString> FolderIds;
-        THashSet<TString> EntityIds;   // DiskIds or FsIds
-        bool IsBlacklist = false;
+        TFilters Whitelist;
+        TFilters Blacklist;
         double CloudProbability = 0;
-        double FolderProbability = 0;
+        double FolderProbability = 0;;
```
Comment (qkrorlqr) [inline cloud/storage/core/libs/features/features_config.h:22]: лишняя ;

File: cloud/storage/core/libs/features/features_config.cpp [hunk 2]
```diff
@@ -17,34 +17,25 @@ bool ProbabilityMatch(double p, const TString& s)
...
-        info.CloudProbability = feature.GetCloudProbability();
-        info.FolderProbability = feature.GetFolderProbability();
-
-        const auto& cloudList = feature.HasBlacklist()
-            ? feature.GetBlacklist() : feature.GetWhitelist();
-
-        for (const auto& cloudId: cloudList.GetCloudIds()) {
-            info.CloudIds.emplace(cloudId);
-        }
+    double defaultProbability = config.HasWhitelist() ? 0.0 : 1.0;
```
Comment (tpashkin) [inline cloud/storage/core/libs/features/features_config.cpp:25]: This is included only for backwards compatibility. Specifically not to break the config that is currently deployed for some clusters: blacklist is set, but no probability is specified

File: cloud/storage/core/libs/features/features_config.h [hunk 3]
```diff
@@ -1,10 +1,10 @@
 #pragma once
 
+#include "filters.h"
```
Comment (sharpeye) [inline cloud/storage/core/libs/features/features_config.h:3]: public.h не должен первым всегда идти?

**Review by qkrorlqr — DISMISSED** [review]

File: cloud/storage/core/libs/features/filters.h [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+#include <cloud/storage/core/config/features.pb.h>
+
+#include <util/generic/hash_set.h>
+#include <util/generic/string.h>
+
+namespace NCloud::NFeatures {
+
+////////////////////////////////////////////////////////////////////////////////
+
+class TFilters {
```
Comment (sharpeye) [inline cloud/storage/core/libs/features/filters.h:12]: { не там

File: cloud/storage/core/libs/features/filters.h [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+private:
+    THashSet<TString> CloudIds;
+    THashSet<TString> FolderIds;
+    THashSet<TString> EntityIds;   // DiskIds or FsIds
+
+public:
+    TFilters(NProto::TFilters config);
```
Comment (sharpeye) [inline cloud/storage/core/libs/features/filters.h:19]: М.б. explicit?

Comment (tpashkin) [reply cloud/storage/core/libs/features/filters.h:20]: Не совсем понимаю зачем он тут

**Review by qkrorlqr — APPROVED** [review]

**Review by sharpeye — APPROVED** [review]

**Pull request merged**